### PR TITLE
Issue #DP-1103 feat: Assign users to a project to be made easier(user search)

### PR DESCRIPTION
**Pull request by santoshilimi**

### CODE REVIEW

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 1]
```diff
@@ -247,6 +252,26 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
...
+      let filteredUser = this.registryService.getSearchedUserList(this.initialSourcingOrgUser, this.searchInput);
+      filteredUser.length > this.searchLimitCount ? this.searchLimitMessage = true: this.searchLimitMessage = false;   
+      this.sortUsersList(filteredUser);
+    } else {
+      this.searchLimitMessage = false;
+      this.sortUsersList(this.initialSourcingOrgUser);
+    }
+  }
+
+  sortUsersList(usersList) {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:267]: @santoshilimi same code is repeated on line no: 282-288, Can you try to use this same function there?

File: src/app/client/src/app/modules/contribute/components/org-user-list/org-user-list.component.ts [hunk 2]
```diff
@@ -100,6 +105,26 @@ export class OrgUserListComponent implements OnInit, AfterViewInit {
...
+      let filteredUser = this.registryService.getSearchedUserList(this.initialSourcingOrgUser, this.searchInput);
+      filteredUser.length > this.searchLimitCount ? this.searchLimitMessage = true: this.searchLimitMessage = false;   
+      this.sortUsersList(filteredUser);
+    } else {
+      this.searchLimitMessage = false;
+      this.sortUsersList(this.initialSourcingOrgUser);
+    }
+  }
+  
+  sortUsersList(usersList) {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/contribute/components/org-user-list/org-user-list.component.ts:120]: @santoshilimi same code is repeated on line no: 139-143, Can you try to use this same function there?

File: src/app/client/src/app/modules/core/services/registry/registry.service.ts [hunk 3]
```diff
@@ -19,6 +19,9 @@ export class RegistryService extends DataService {
   public http: HttpClient;
   public mycontributionOrgUsers = [];
   osReqLimit =  250;
+  searchLimitCount = 15;// need to chnage once dev testing done (100) and in future we can get/set limit from configuration
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/core/services/registry/registry.service.ts:22]: @santoshilimi lets not forget to change this

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts (hunk 1 above)
Comment (santoshilimi) [reply src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts]: resolved

File: src/app/client/src/app/modules/contribute/components/org-user-list/org-user-list.component.ts (hunk 2 above)
Comment (santoshilimi) [reply src/app/client/src/app/modules/contribute/components/org-user-list/org-user-list.component.ts]: resolved

File: src/app/client/src/app/modules/core/services/registry/registry.service.ts (hunk 3 above)
Comment (santoshilimi) [reply src/app/client/src/app/modules/core/services/registry/registry.service.ts:22]: Sure

File: src/app/client/src/app/modules/program/components/org-user-list/org-user-list.component.spec.data.ts [hunk 4]
```diff
@@ -0,0 +1,324 @@
+export const userDetail =
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/program/components/org-user-list/org-user-list.component.spec.data.ts:1]: If this is the same test data used across all four components, shall we put it in one file and share it with the rest?

Comment (santoshilimi) [reply src/app/client/src/app/modules/program/components/org-user-list/org-user-list.component.spec.data.ts]: Resolved

**Pull request merged**